pages/pipelines/secrets.md.erb

```diff
@@ -11,57 +11,56 @@ A best practice for secret storage is to use your own secrets storage service, s
-your agent machine, are only accessible by you, and can conditionally expose
-secrets to your pipeline steps.
+If you don't use a secrets storage service, then you can use the Buildkite agent's `environment` hook to export secrets to a job.
 
-For example, say you had the following deploy [command step](/docs/pipelines/command-step) in a pipeline:
+The `environment` hook is a shell script that is sourced at the beginning of a job.
+It runs within the job's shell, so you can use it to conditionally run commands and export secrets within the job.
 
-```yml
-steps:
-  - command: scripts/trigger-github-deploy
-    key: trigger-github-deploy
-```
-{: codeblock-file="pipeline.yml"}
+By default, the `environment` hook file is stored in the agent's `hooks` directory.
+The path to this directory varies by platform; read the [installation instructions](/docs/agent/v3/installation) for the path on your platform.
+The path can also be overridden by the [`hooks-path`](/docs/agent/v3/hooks#hook-locations-agent-hooks) setting.
 
-This step runs the `scripts/trigger-github-deploy` script from the repository, and creates a GitHub Deployment using a [GitHub personal access token](https://help.github.com/en/articles/creating-a-personal-access-token-for-the-command-line). The script has the following source code:
+For example, suppose you want to expose a Test Analytics API token to a specific pipeline.
+To do this, create an `environment` script in your agent's `hooks` directory that checks for the pipeline slug before exporting the secret:
 
 ```bash
 #!/bin/bash
 set -euo pipefail
 
-curl \
-  --header "Authorization: token $GITHUB_MY_APP_DEPLOYMENT_ACCESS_TOKEN" \
-  --request POST \
-  --data "{\"ref\": \"$BUILDKITE_COMMIT\"}" \
-  https://api.github.com/repos/my-org/my-app/deployments
+if [[ "$BUILDKITE_PIPELINE_SLUG" == "pipeline-one" ]]; then
+  export BUILDKITE_ANALYTICS_TOKEN="oS3AG0eBuUJMWRgkRvek"
+fi
 ```
-{: codeblock-file="scripts/trigger-github-deploy"}
+{: codeblock-file="hooks/environment"}
 
-To set up the `GITHUB_MY_APP_DEPLOYMENT_ACCESS_TOKEN` secret for the step, you
-could define an [`environment` hook](/docs/agent/v3/hooks#job-lifecycle-hooks)
-in your [agent `hooks-path` directory](/docs/agent/v3/hooks#hook-locations-agent-hooks)
-to conditionally export it. For example:
+Adding conditional checks, such as the pipeline slug and step identifier, helps to limit accidental disclosure of secrets.
+For example, suppose you have a step that runs a script expecting a `SECRET_DEPLOYMENT_ACCESS_TOKEN` environment variable, like this one:
+
+```yml
+steps:
+  - command: scripts/trigger-deploy
+    key: trigger-deploy
+```
+{: codeblock-file="pipeline.yml"}
+
+In your `environment` hook, you can export the deployment token only when when the job is the deployment step in a specific pipeline:
 
 ```bash
 #!/bin/bash
 set -euo pipefail
 
 if [[ "$BUILDKITE_PIPELINE_SLUG" == "my-app" ]]; then
```

Comment:
Would it make more sense to chain conditions using && instead of nesting using if? It fits my brain better, but I don't feel strongly about this.

Reply:
The original was nested and I thought it might've been a house style thing. Switched to logical AND with 016a43cb.